### DISCUSSION TITLE
Override type of column

### DISCUSSION THREAD

**Discussion by gabegorelick** [hasura/graphql-engine#8620] [Question] [answered]
Is there any way to override the type of a column at the Hasura level without altering the underlying DB? 

My use case: I have a bigint column in Postgres that I would like to be treated as `ID` at the GQL layer and serialized as a string. Judging by https://github.com/hasura/graphql-engine/issues/2881, it seems like I cannot map that bigint column to an `ID` GQL type. All I can do is set `HASURA_GRAPHQL_STRINGIFY_NUMERIC_TYPES` globally so that *all* bigints are serialized as strings. Or `ALTER` the underlying column at the DB level to be a string, but that's a very heavyweight change.

Is there any way to tell Hasura "serialize this specific column as a string"?

Other things I've tried:
- Add a computed field that casts the bigint to the correct type. This would require updating consumers to point to the new field.  As far as I can tell, there's no way to mask a field with a computed one (Hasura rejects the metadata update if you try that).

**Comment by BenoitRanque** [comment, marked as answer]
It seems you are asking for two things:

# Ability to arbitrarily map column to custom scalar type
This is not currently possible. If this feature is important to you, you may open a feature request.

# Ability to arbitrarily cast a column to another type
There is no dedicated setting for this, however it can be achieved with the following workaround:

- [Rename](https://hasura.io/docs/latest/graphql/core/databases/postgres/schema/custom-field-names/#introduction) the original `id` column to something like `_id`
- Add a computed field that casts the bigint to the correct type, and name it `id`

I'm of course assuming your column to be mapped is named `id`, change this as required.

**Reply by gabegorelick** [reply]
> Rename the original id column to something like _id

Just a note on this: you can't use a custom GQL field name to do this. Instead, you have to rename the column at the DB level. This is because the new computed field can't have a name that conflicts with a DB column (Hasura throws an error preventing this if you try to configure it).

**Reply by adas98012** [reply]
Thanks for the additional details. We have created a new Github [issue](https://github.com/hasura/graphql-engine/issues/8699) for this. You should now be able to track the progress of this ticket.
